### ISSUE TITLE
Ordering of @ApiObjectField

### ISSUE THREAD

**Issue by imade** [fabiomaffioletti/jsondoc#104]
Is there a way or workaround to specify the order of the ApiObjectFields? Problem is that if I have a following class as a POST body object:
```Java
package com.yt.mps.cmdrs.domain.sendsms;

import org.jsondoc.core.annotation.ApiObject;
import org.jsondoc.core.annotation.ApiObjectField;

@ApiObject
public class Person {

    @ApiObjectField
    private String name;

    @ApiObjectField
    private String email;

    //...
}
```

Then in Jsondoc UI playground the fields appear in alphabetical order:
```JSON
{
  "email": "",
  "name": ""
}
```

This is a simplistic example, but In real world with more complex structures being able to list the fields in logical order makes the API much easier to understand for the clients.

**Comment by fabiomaffioletti** [comment]
This could be an interesting improvement. The only way I can think of to implement it is to add an additional property "order" to the ApiObjectField annotation or another annotation "ApiOrder" to be placed on the field and use the value in a comparator. What do you think?

**Comment by imade** [comment]
I think we could have 2 ways:
1. By convention - without explicitly specifying the order use the same sequence as pojo properties are defined in.
2. Explicit - yes, something like @ApiObjectField(order = 1)

**Comment by fabiomaffioletti** [comment]
Number one can't be done: http://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getDeclaredFields()

> Returns an array of Field objects reflecting all the fields declared by the class or interface represented by this Class object. This includes public, protected, default (package) access, and private fields, but excludes inherited fields. The elements in the array returned are not sorted and are not in any particular order.

**Comment by imade** [comment]
OK, I could live without number 1, it's more of a convenience aspect.

**Comment by bassmake** [comment]
Hi, any news in this issue?

**Comment by fabiomaffioletti** [comment]
Not yet. Don't have time at the moment. Submit a pull request if you can.

**Comment by bassmake** [comment]
I started to work on this.
Do you think that ApiObjectFieldDoc.order should be int or Integer? (-1 or null in comparator as default value)
Also I think that fields with no order should be generated after fields with order. Agree?

**Comment by fabiomaffioletti** [comment]
Thank you.
- should be Integer. I would say that -1 (-2, -3, negative values in general) needs to be considered as a valid value, so a field annotated with order = -1 needs to be shown before the one annotated with order = 0. This would give more flexibility. Default value for the order property should be Integer.MAX, so that in the comparator it will be considered as the last in the list.
- a new class JSONDocTemplate could be fine, but if you find that it's not easy to convert the Map to a class, then to keep the field order you should consider having a more convenient Map implementation like LinkedHashMap or TreeMap.
What do you think? Please remember to add unit tests ;)

**Comment by fabiomaffioletti** [comment]
Released with version 1.1.15
